pkg/kudoctl/util/repo/index.go

```diff
@@ -48,15 +48,32 @@ func (b PackageVersions) Swap(i, j int) { b[i], b[j] = b[j], b[i] }
 // This is needed to allow sorting of packages.
 func (b PackageVersions) Less(x, y int) bool {
 	// Failed parse pushes to the back.
-	i, err := semver.NewVersion(b[x].Version)
+	appVersionX, err := semver.NewVersion(b[x].AppVersion)
 	if err != nil {
 		return true
 	}
-	j, err := semver.NewVersion(b[y].Version)
+
+	appVersionY, err := semver.NewVersion(b[y].AppVersion)
 	if err != nil {
 		return false
 	}
-	return i.LessThan(j)
+
+	if appVersionX.Equal(appVersionY) {
+		// Failed parse pushes to the back.
+		versionX, err := semver.NewVersion(b[x].OperatorVersion)
+		if err != nil {
+			return true
+		}
+
+		versionY, err := semver.NewVersion(b[y].OperatorVersion)
+		if err != nil {
+			return false
+		}
+
+		return versionX.LessThan(versionY)
+	}
+
+	return appVersionX.LessThan(appVersionY)
```

Comment:
I feel we should have a number of tests around this.   This is more complex than it was originally.

Reply:
I still feel this is true... but approving with the assumption that it will happen or will have an issue created to do it.

Reply:
I added some additional test cases to `TestParseIndexFile` in `index_test.go`. But looking at it again, it doesn't task _this_ particular code path. I'll add more entries to the test index to make sure that every case is covered.